jax/_src/scipy/linalg.py

```diff
@@ -1041,3 +1043,46 @@ def toeplitz(c: ArrayLike, r: Optional[ArrayLike] = None) -> Array:
       (nrows,), (1,), 'VALID', dimension_numbers=('NTC', 'IOT', 'NTC'),
       precision=lax.Precision.HIGHEST)[0]
   return jnp.flip(patches, axis=0)
+
+
+_NULLSPACE_DOC = """\
+Because the size of the output of ``null_space`` is data-dependent, the function is not
+typically compatible with JIT. The JAX version adds the optional ``size`` argument which
+must be specified statically for ``jax.scipy.linalg.null_space`` to be used within some of JAX's
+transformations.
+"""
+_NULLSPACE_EXTRA_PARAMS = """
+size : int, optional
+    If specified, ``size`` elements of the null space are returned. If there are
+    fewer null space elements than ``size`` indicates, the return value will be padded with constant
+    0 vectors that belong by nature to the null space.
+"""
+
+@_wraps(scipy.linalg.null_space, lax_description=_NULLSPACE_DOC, extra_params=_NULLSPACE_EXTRA_PARAMS)
+def null_space(A: ArrayLike, rcond: Optional[Union[float, Array]] = None,
+               size: Optional[int] = None) -> Array:
+  u, s, vh = svd(A, full_matrices=True)
+  M, N = u.shape[0], vh.shape[1]
+  if rcond is None:
+    rcond = jnp.finfo(s.dtype).eps * max(M, N)
+  rcond = jnp.array(rcond).astype(s.dtype)
+  tol = jnp.amax(s) * rcond
+  n_null_space = vh.shape[0] - jnp.sum(s > tol, dtype=int)
+  if size is None:
+    size = n_null_space
+  size = core.concrete_or_error(operator.index, size,
```

Comment:
Maybe something like this:
```python
size_int = core.concrete_or_error(operator.index, n_null_space if size is None else size,
  ...
```
then use `size_int` below.

Reply:
seems to be working : sorry I am really not used to typing so I don't know exactly how to deal with it in a straightforward way.